Source/WebKit/WebProcess/WebPage/Cocoa/UnifiedTextReplacementController.mm

```diff
@@ -459,6 +459,73 @@ static void replaceContentsInRange(WebCore::LocalFrame& frame, const WebCore::Si
         return;
     }
 
+    auto replacementType = m_replacementTypes.get(uuid);
+    switch (replacementType) {
+    case WebUnifiedTextReplacementType::PlainText:
+        return textReplacementSessionPerformEditActionForPlainText(*document, uuid, action);
+    case WebUnifiedTextReplacementType::RichText:
+        return textReplacementSessionPerformEditActionForRichText(*document, uuid, action);
+    }
+}
+
+void UnifiedTextReplacementController::textReplacementSessionPerformEditActionForPlainText(WebCore::Document& document, const WTF::UUID& uuid, WebKit::WebTextReplacementData::EditAction action)
+{
+    RELEASE_LOG(UnifiedTextReplacement, "UnifiedTextReplacementController::textReplacementSessionPerformEditActionForPlainText (%s) [action: %hhu]", uuid.toString().utf8().data(), enumToUnderlyingType(action));
+
+    RefPtr liveRange = m_contextRanges.get(uuid);
+    auto sessionRange = makeSimpleRange(liveRange);
+    if (!sessionRange) {
+        ASSERT_NOT_REACHED();
+        return;
+    }
+
+    Vector<Ref<WebCore::Node>> sessionNodes;
+    for (Ref node : intersectingNodes(*sessionRange))
+        sessionNodes.append(node);
+
+    for (auto nodeIterator = sessionNodes.rbegin(); nodeIterator != sessionNodes.rend(); ++nodeIterator) {
+        auto node = *nodeIterator;
+        auto markers = document.markers().markersFor(node, { WebCore::DocumentMarker::Type::UnifiedTextReplacement });
+
+        for (auto markerIterator = markers.rbegin(); markerIterator != markers.rend(); ++markerIterator) {
+            auto marker = *markerIterator;
```

Comment:
Should this check if the `WeakPtr` is null and skip if so? Because we're triggering editing commands and layout in this loop, I suspect it's possible that some of the (previously non-null) markers in this vector could become null during iteration.

Reply:
Will do.